CONTRIBUTING.adoc

```diff
@@ -0,0 +1,46 @@
+= Contributing
+// settings:
+:idprefix:
+:idseparator: -
+:source-language: ruby
+:language: {source-language}
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+// URIs:
+:uri-repo: https://github.com/asciidoctor/asciidoctor
+:uri-issues: {uri-repo}/issues
+:uri-squash-help: https://chris.beams.io/posts/git-commit/#seven-rules
+:uri-gist: https://gist.github.com
+
+== License Agreement
+
+By contributing changes to this repository, you agree to license your contributions under the MIT license.
+This ensures your contributions have the same license as the project and that the community is free to use your contributions.
+You also assert that you are the original author of the work that you are contributing unless otherwise stated.
+
+== Submitting an Issue
+
+We use the {uri-issues}[issue tracker on GitHub] associated with this project to track bugs and features.
+Before submitting a bug report or feature request, check to make sure it hasn't already been submitted.
+When submitting a bug report, please include a {uri-gist}[Gist] that includes any details that may help reproduce the bug.
+
+== Submitting a Pull Request
+
+. Fork the repository.
+. Build the project `gradlew build`.
+. Before commits or pushes run `gradlew -i codenarcAll`
+  - If there are errors fix them
+  - It may be the case where the errors or warnings are necessary. If that is the case, ask before ignoring them
+. If you made multiple commits make sure you squash the commits
```

Comment:
Personal style obsession here, for consistency all sentences should end with a dot.